src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -4565,7 +4565,23 @@ Qgis::VectorExportResult QgsPostgresProvider::createEmptyLayer( const QString &u
 
     // get geometry type, dim and srid
     int dim = 2;
-    long srid = srs.postgisSrid();
+    long srid;
+
+    // get srid from PostGIS
+
+    if ( !srs.authid().isEmpty() )
+    {
+      QStringList authId_list = srs.authid().split(':');
+
+      sql = QStringLiteral( "SELECT srid FROM public.spatial_ref_sys WHERE auth_name = %1 AND auth_srid = %2" )
+              .arg( quotedValue( authId_list[0] ) )
+              .arg( quotedValue( authId_list[1] ) );
+
+      result = conn->LoggedPQexec( "QgsPostgresProvider", sql );
+      if ( result.PQresultStatus() != PGRES_TUPLES_OK )
+        throw PGException( result );
+      srid = result.PQgetvalue( 0, 0 ).toLong();
```

Comment:
We'll need to check whether there IS a match, and handle accordingly:

```suggestion
    if ( result.PQntuples() < 1 )
    {
      if ( errorMessage )
      *errorMessage = QObject::tr( "Creation of data source %1 failed: \n No record matching %2 found in public.spatial_ref_sys" )
                        .arg( schemaTableName, srs.authid() );

    conn->LoggedPQexecNR( "QgsPostgresProvider", QStringLiteral( "ROLLBACK" ) );
    conn->unref();
    return Qgis::VectorExportResult::ErrorCreatingLayer;

    srid = result.PQgetvalue( 0, 0 ).toLong();
```